### ISSUE TITLE
MIDI output to M-Audio USB Uno MIDI Interface to Yamaha Keyboard.  Works in MS3, not in MS4

### ISSUE THREAD

**Issue by zacjansheski** [musescore/MuseScore#19337]
### Issue type

General playback bug

### Bug description

MIDI output to M-Audio USB Uno MIDI Interface to Yamaha Keyboard.  Works in MS3, not in MS4

### Steps to reproduce

1. Connect MIDI devices
2. Set output to USB Uno MIDI Interface
3. Works in MS3, not in MS4

### MuseScore Version

4.1.1

### Regression

Yes, this used to work in MuseScore 3.x and now is broken

### Operating system

OS: macOS 11.6

**Comment by zacjansheski** [comment]
MS3 I/O settings (sound plays out of Yamaha keyboard)

<img width="899" alt="Screen Shot 2023-09-08 at 12 34 44 PM" src="https://github.com/musescore/MuseScore/assets/69917893/d672ce7d-721e-4200-bb21-46872b501c4a">

MS4 I/O settings (Sound does not play out of Yamaha keyboard)

<img width="874" alt="Screen Shot 2023-09-08 at 12 34 53 PM" src="https://github.com/musescore/MuseScore/assets/69917893/2ed922bb-df99-4177-98c3-7fd55075c24d">

**Comment by marcelser** [comment]
I'm a bit puzzled that this bug about no MIDI output is treated as P3 Low. It makes MU4 unusable for the time beeing for people who are affected. And it seems it's cross device as my bug that was closed was windows and this one is MacOS

**Comment by zacjansheski** [comment]
I'll pump the priority since it appears to be more widespread.

I wasn't sure if it was just my devices at the time of logging.

**Comment by cbjeukendrup** [comment]
Note that MIDI output is only active whenever MS Basic is selected as the sound in the Mixer. (Yes, this is weird, but there is already at least one open issue for that.)
Is that why it seems to be non-functional?
And does that MIDI 2.0 checkbox in Preferences make any difference?

**Comment by marcelser** [comment]
Doesn't make any difference. Tried various mscz files setting all tracks to MS Basic (Automatic) also tried MS Basic (Piano) but doesn't create any MIDI Output at all. And on windows there's no Midi 2.0 Option so I can't try that (maybe the OP can try this). So it's not related to MS Basic. It's something else that's broken in MIDI Output and it's even across different OSses

**Comment by aru7777** [comment]
I have the same issue on Linux for my Yamaha Motif XF8 keyboard. Will this be fixed soon? I can't install MuseScore3 as an alternative.

**Comment by MarcSabatella** [comment]
Can you explain in more detail what goes wrong?  Are you saying the device doens't show up in the Audio & MIDI section of Edit / Preferences?  Or that it shows up but does not respond to MIDI the output that is sent?  Are you making sure to use the current verison 4.5.2, the official AppImage and not any sort of unsupported third party build?  Did you also check the Mixer to be sure you aren't trying to use MuseSounds with any of the tracks you are trying to play>

**Comment by aru7777** [comment]
Yes, I'm using the official version, I tried a lot of them with no luck. Every other software lists 4 midi channels for my keyboard, the first one is the one I need. MuseScore only lists one with the name "Yamaha Motif XF8". Selecting it doesn't work for either midi input nor output. I'm not trying to use MuseSounds and every other setting is at default value.

**Comment by MarcSabatella** [comment]
Did you verify you are using 4.5.2?  Some earlier versions sent notes with low velocity settings and instead used CC messages for volume control, which caused many keyboard to not produce any audible results even though the data was being sent.  The current version should be sending velocity information in the more traditional fashion.  Have you tried with MIDI 2.0 turned on as well as off in the Preferences?

**Comment by aru7777** [comment]
I'm on version 4.52 Appimage from Github. I checked but could not find a Midi 2.0 setting in the Preferences. It's not just the audio, there is no midi input either.

**Comment by aru7777** [comment]
I prefer using the flatpak version because AppImages aren't supported natively on OpenSuse AEON but for the sake for trying the official way, I did whatever was needed to run the official AppImage. The issue is still there. So I went back to the flatpak version since there is no benefit for me to use the AppImage binary.

Anyway, I found a workaround: there is an app called [VMPK](
https://vmpk.sourceforge.io/) (Virtual Midi Piano Keyboard). It's normally used as a virtual keyboard but you can connect an actual MIDI keyboard to it and redirect the MIDI output to Midi Through.

I had a strange bug at first where notes where played indefinitely but fixed it somehow. Anyway, using MidiThrough and VMPK, MuseScore detects MIDI Input. Not sure if it can be used the same way for Midi Output though, at least not without messing with the settings back and forth.

It would be nice if an actual fix could be implemented in MuseScore though. For what it's worth, MuseScore 3 detected 4 Midi Channels for my keyboard when MS4 only detects one. Could be related or not, I don't know.
